fe/fe-core/src/main/java/org/apache/doris/analysis/CreateFunctionStmt.java

```diff
@@ -175,14 +175,19 @@ public Expr getOriginFunction() {
     public void analyze(Analyzer analyzer) throws UserException {
         super.analyze(analyzer);
 
-        analyzeCommon(analyzer);
-        // check
-        if (isAggregate) {
-            analyzeUda();
-        } else if (isAlias) {
-            analyzeAliasFunction();
-        } else {
-            analyzeUdf();
+        // https://github.com/apache/doris/issues/17810
+        // this error report in P0 test, so we suspect that it is related to concurrency
+        // add this change to test it.
+        synchronized (CreateFunctionStmt.class) {
```

Comment:
If we release doris 2.0 now, this code will be released. It may cause FE analysis time very long....

I think you could using Config.use_fuzzy_session_variable to control that the sychronized logic only work in github workflow.